src/pushsource/_impl/backend/koji_source.py

```diff
@@ -283,18 +284,38 @@ def _push_items_from_rpm_meta(self, rpm, meta):
 
         candidate_paths = []
 
-        # If signing keys requested, try them in order of preference
-        for key in self._signing_key:
-            if key:
-                key = key.lower()
-                candidate = os.path.join(build_path, self._pathinfo.signed(meta, key))
-            else:
-                candidate = unsigned_path
-            candidate_paths.append(candidate)
-            if os.path.exists(candidate):
-                rpm_path = candidate
-                rpm_signing_key = key
+        timeout = int(os.getenv("PUSHSOURCE_SRC_POLL_TIMEOUT") or "0")
+        poll_rate = int(os.getenv("PUSHSOURCE_SRC_POLL_RATE") or "30")
+        max_attempts = timeout // poll_rate
+
+        for i in range(max_attempts + 1):
+            # We only want to poll if signing key is requested
+            if not self._signing_key:
                 break
+            # If signing keys requested, try them in order of preference
+            for key in self._signing_key:
+                if key:
+                    key = key.lower()
+                    candidate = os.path.join(
+                        build_path, self._pathinfo.signed(meta, key)
+                    )
+                else:
+                    candidate = unsigned_path
+                if candidate not in candidate_paths:
+                    candidate_paths.append(candidate)
+                if os.path.exists(candidate):
+                    rpm_path = candidate
+                    rpm_signing_key = key
+                    break
+
+            if rpm_path or i == max_attempts:
+                break
+            LOG.info(
+                "Waiting for %s seconds for any of the following paths to appear: %s",
+                poll_rate,
+                ", ".join(candidate_paths),
+            )
+            time.sleep(poll_rate)
 
```

Comment:
The code actually raises an error if the path is not present. This is why I didn't add a warning, because the code will have an error message regardless. 

It can be seen at http://10.0.149.4/pub/task/15857/log/stdout.log

Reply:
Hmm, that error comes from ErrataSource though, which is not the only way KojiSource is used.

Reply:
After separating the polling to its own function, there's now a warning message.